articles/extensions/github-deploy.md

```diff
@@ -0,0 +1,116 @@
+
+The __Deploy__ button lets you manually deploy the rules and database connection scrips you already have in your GitHub repository. This is useful if you already have a repository filled with scripts and you want to deploy them once you setup the extension. It can also come in handy in case you accidentally delete some things in Auth0 and you want to redeploy the latest version of your repository again.
+
+::: panel-warning Full redeploy
+The full redeployment means that any rules or database connection scripts that exist in Auth0 but don't exist in your GitHub repository will be __deleted__, because your GitHub repository becomes the single source of truth.
+:::
+
+
+### Deploy database connection scripts
+
+In order to start deploying database connection scripts, first you have to create a directory under `database-connections`. The name of the directory must match __exactly__ the name of your [database connection](${uiURL}/#/connections/database) in Auth0. Of course you can create as many directories as your database connections.
+
+Under the created directory, create one file for every script you want to use. The allowed scripts are:
+
+- `get_user.js`
+- `create.js`
+- `verify.js`
+- `login.js`
+- `change_password.js`
+- `delete.js`
+
+Only the `login.js` script is required in a custom database connection. 
+
+If you enabled the migration feature, you'll also need to provide the `get_user.js` script.
+
+You can find an example in [this GitHub repository](https://github.com/auth0-samples/github-source-control-integration/tree/master/database-connections/my-custom-db).
+
+
+### Deploy rules
+
+In order to deploy a rule, you have to create a JavaScript file under the `rules` directory of your GitHub repository. Each rule must be on its own `.js` file. 
+
+For example, if you create the file `rules/set-country.js`, then the extension will create a rule in Auth0 with the name `set-country`. 
+
+__NOTE__: If you plan to use Source Control integration for an existing account, make sure you rename your rules in Auth0 first to the same name of the files you'll deploy to this directory.
+
+Moreover, you can control the rule order, status (`enabled`/`disabled`) and stage (`login_success`, `login_failure`, `user_registration`), by creating a JSON file next to your JavaScript file, using the same name. So for this example you would create a file named `rules/set-country.json`.
+
+__set-country.js__
+```javascript
+function (user, context, callback) {
+  if (context.request.geoip) {
+    user.country = context.request.geoip.country_name;
+  }
+  callback(null, user, context);
+}
+```
+
+__set-country.json__
+```javascript
+{
+  "enabled": false,
+  "order": 15,
+  "stage": "login_success"
+}
+```
+
+Having multiple rules with the same order is not allowed, so make sure you don't have any collisions. Also, keep in mind that for the `user_registration` and `login_failure` stage only a single rule is allowed.
```

Comment:
Maybe add a note with best practices? Like: "We suggest you to create a json file for every rule and explicitly assign an order to each rule. If then you leave enough space between rules it will be easy to move them around without causing collisions. Eg:

 Order 10 - log_user.js
 Order 20 - set_country.js
 Order 30 ...
"